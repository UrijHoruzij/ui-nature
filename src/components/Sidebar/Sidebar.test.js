import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { Sidebar, Avatar, Typography, Button } from "../";
import ThemeProvider from "../../ThemeProvide";

describe("Sidebar component", () => {
  test("Render sidebar", () => {
    const component = (
      <ThemeProvider>
        <Sidebar
          blur
          color="rgba(10,110,168,0.25)"
          background="https://images.unsplash.com/photo-1605505027420-a74af00e6ee2"
        >
          <Typography type="h1">Хранилище</Typography>
          <Avatar
            avatar="https://images.unsplash.com/photo-1517841905240-472988babdf9"
            fullname="Helena Lopes"
            style={{ marginTop: "24px" }}
            size="large"
          />
          <Button style={{ marginTop: "24px" }}>Загрузить</Button>
        </Sidebar>
      </ThemeProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
