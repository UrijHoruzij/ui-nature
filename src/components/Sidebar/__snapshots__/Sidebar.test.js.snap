// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component Render sidebar 1`] = `
.c7 {
  font-family: Montserrat,sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  margin: 0;
  padding: 8px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 32px;
  border: none;
  border-radius: 4px;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
  box-sizing: border-box;
  color: #ffffff;
  background-color: #008603;
}

.c7:hover {
  background-color: #339e35;
}

.c3 {
  font-family: Montserrat,sans-serif;
  border: none;
  outline: none;
  font-weight: 500;
  color: #333333;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  font-size: 24px;
  line-height: 32px;
}

.c0 {
  width: 320px;
  max-width: 320px;
  height: 100%;
  max-height: 100vh;
  background-repeat: no-repeat;
  background-size: cover;
  z-index: 0;
  background-image: url(https://images.unsplash.com/photo-1605505027420-a74af00e6ee2);
}

.c1 {
  width: 100%;
  height: 100%;
  z-index: 1;
  background-color: rgba(10,110,168,0.25);
}

.c2 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  z-index: 2;
  -webkit-backdrop-filter: blur(16px);
  backdrop-filter: blur(16px);
}

.c5 {
  position: relative;
  display: inline-block;
  width: 176px;
  height: 176px;
}

.c6 {
  border-radius: 50%;
  overflow: hidden;
  width: 100%;
  height: 100%;
  object-fit: cover;
  z-index: 5;
  max-width: 176px;
  max-height: 176px;
}

<div
  className="c0"
  color="rgba(10,110,168,0.25)"
>
  <div
    className="c1"
    color="rgba(10,110,168,0.25)"
  >
    <div
      className="c2"
    >
      <h1
        className="c3 c4"
        color="text"
        type="h1"
      >
        Хранилище
      </h1>
      <div
        className="c5"
        size="large"
        style={
          Object {
            "marginTop": "24px",
          }
        }
      >
        <img
          alt="Helena Lopes"
          className="c6"
          size="large"
          src="https://images.unsplash.com/photo-1517841905240-472988babdf9"
        />
      </div>
      <button
        className="c7"
        style={
          Object {
            "marginTop": "24px",
          }
        }
        type="primary"
      >
        Загрузить
      </button>
    </div>
  </div>
</div>
`;
